src/org/exist/xmldb/LocalXMLResource.java

```diff
@@ -190,6 +197,42 @@ public Node getContentAsDOM() throws XMLDBException {
                 }
             });
         }
+
+        return exportInternalNode(result);
+    }
+
+    /**
+     * Provides a safe export of an internal persistent DOM
+     * node from eXist via the Local XML:DB API.
+     *
+     * This is done by providing a proxy object that only implements
+     * the appropriate W3C DOM interface. This helps prevent the
+     * XML:DB Local API from leaking implementation through
+     * its abstractions.
+     */
+    private Node exportInternalNode(final Node node) {
```

Comment:
no because of 192 (202) ; will there be a scenario with a NULL value for `result` ?